application/forms/EventRuleConfigForm.php

```diff
@@ -411,7 +409,7 @@ public function getValues(): array
     protected function createRemoveButton(string $prefix): SubmitButtonElement
     {
         /** @var array<int, array<string, mixed>> $escalations */
-        $escalations = $this->config['rule_escalation'] ?? [];
+        $escalations = $this->getValues()['rule_escalation'];
```

Comment:
The `getValues()` method is called too often and performs a query each time. This should cache the last changes to avoid the unnecessary process.